ibm/service/vpc/resource_ibm_is_bare_metal_server.go

```diff
@@ -2388,32 +2389,79 @@ func bareMetalServerUpdate(context context.Context, d *schema.ResourceData, meta
 				if allowIpSpoofing {
 					bmsNicPatchModel.AllowIPSpoofing = &allowIpSpoofing
 				}
+				nicflag = true
 			}
 		}
 		if d.HasChange("primary_network_interface.0.enable_infrastructure_nat") {
 			if enableNatOk, ok := d.GetOk("primary_network_interface.0.enable_infrastructure_nat"); ok {
 				enableNat := enableNatOk.(bool)
 				bmsNicPatchModel.EnableInfrastructureNat = &enableNat
+				nicflag = true
+			}
+		}
+		if d.HasChange("primary_network_interface.0.security_groups") && !d.IsNewResource() {
```

Comment:
This change doesnt require to set  				nicflag = true

Reply:
No, adding of sg groups is taken care inside this block itself.